lib/datasource/go/index.ts

```diff
@@ -113,13 +123,27 @@ export async function getReleases({
 }: GetReleasesConfig): Promise<ReleaseResult | null> {
   logger.trace(`go.getReleases(${lookupName})`);
   const source = await getDatasource(lookupName);
-  if (source?.datasource !== github.id && source?.datasource !== gitlab.id) {
-    return null;
+  let res = null;
+
+  switch (source.datasource) {
+    case github.id: {
+      res = await github.getReleases(source);
+      break;
+    }
+    case gitlab.id: {
+      res = await gitlab.getReleases(source);
+      break;
+    }
+    case bitbucket.id: {
+      res = await bitbucket.getReleases(source);
+      break;
+    }
+    /* istanbul ignore next */
+    default: {
+      return null;
```

Comment:
Please add a test

Reply:
I don't think it is possible to get there, but I had to add something as `default` case.